Fix #19872 :  Responsiveness in the about section was inaccurate.

rahat2134 opened the pull request:
## Overview
1. This PR fixes or fixes part of #19872 .
2. This PR does the following: Responsiveness in the about section was inaccurate in `access the android` button specially. It was not visible at some widths. Resolve all those issues.

## Essential Checklist

- [X] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [X] I have followed the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- [X] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I don't have permissions to assign reviewers directly).
- [X] The linter/Karma presubmit checks pass on my local machine, and my PR follows the [coding style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide)).
- [X] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)

## Proof that changes are correct
[screen-capture (20).webm](https://github.com/oppia/oppia/assets/136263179/17fea14c-1837-4715-8c9b-8fe96a247868)
## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

rahat2134 committed:
Solve-Responsiveness-issue-on-about-us-page

rahat2134 committed:
Merge branch 'develop' of https://github.com/rahat2134/oppia into About-page-overflow

rahat2134 committed:
lint-pass

rahat2134 committed:
lintpass

rahat2134 committed:
lint-pass

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 commented:
@nikitaevg PTAL.

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 commented:
@nikitaevg PTAL.

rahat2134 commented:
@Nik-09 PTAL untill @nikitaevg is back. I think he is busy.

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

nikitaevg reviewed (changes requested):
I don't think it's a good approach to fix the issue. Let's fix it properly. Why the padding or padding-top should be different for 960px, 868px, 698px etc? We have a simple task - the button should be within this container for all widths. Please find a solution that fixes the task without so many if-s, this is a too dirty solution.

rahat2134 commented:
@nikitaevg there was a need for all this padding-top and padding because at each point it was unbalanced; in some areas, there was too much and in others, not enough, which needs to be addressed.
For eg -> 
[screen-capture (19).webm](https://github.com/oppia/oppia/assets/136263179/dc26ec45-c656-422f-9e44-e49471f1bae7)
Here in this video i showed the comparison, if i solve the issue of button only then yes it can be done but since the padding-top was not accurate in most of the conditions. Also by default there were many @media querries present in `about-page.component.css` I just added 2-3 more which were missing or needed to be added. PTAL and tell me if yet need to update it.

hrshkshri commented on core/templates/pages/about-page/about-page.component.css:
```diff
@@ -313,7 +324,8 @@
   }
   .oppia-about-page-section-one .oppia-about-title {
-    padding-top: 10vh;
+    font-size: 25px;
```
Why do we need to change the font size...? 
Also, IG it clashes with `.about-page .oppia-about-section-title `  [here](https://github.com/oppia/oppia/pull/19887/files#diff-eb488cda46641af1186c47c57f144ff50667a490298706beee231c66fc4b8777L302)

rahat2134 replied:
updated this also. Thank You.

hrshkshri commented on core/templates/pages/about-page/about-page.component.css:
```diff
@@ -281,6 +281,17 @@
     height: 50vh;
   }
 }
+@media (orientation: landscape) and (max-width: 960px) {
```
I don't see a point in adding this media query, if you want to apply any changes at `max-width: 960px` you can apply at `max-width: 971px` ([here](https://github.com/oppia/oppia/pull/19887/files#diff-eb488cda46641af1186c47c57f144ff50667a490298706beee231c66fc4b8777L277).
Not much difference between 960px and 971px why create a new media query for that ? : )

rahat2134 replied:
Actually, yes! I tried to add more clarity, but yeah, you are right. We can do that as well. Updated.

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
some final touch

rahat2134 committed:
final touch

rahat2134 committed:
Merge branch 'develop' of https://github.com/oppia/oppia into About-page-overflow

rahat2134 commented on core/templates/pages/about-page/about-page.component.css:
```diff
@@ -324,11 +335,12 @@
-@media (orientation: landscape) and (max-width: 740px) {
-  .oppia-about-page-section-one,
-  .oppia-about-page-section-one .oppia-about-study-image img {
-    height: 90vh;
```
Here, I removed the height because I changed it to 50vh, which is needed to maintain consistency throughout different screen widths.

rahat2134 committed:
remove padding-top

rahat2134 committed:
add padding:0 because there was too much spaces between btns in mobiles

rahat2134 committed:
remove unwanted font-size as wanted by reviewer

rahat2134 commented on core/templates/pages/about-page/about-page.component.css:
```diff
@@ -431,7 +442,7 @@
   .oppia-about-page-section-one .oppia-about-button-container {
     margin: 0;
-    padding-bottom: 5vh;
+    padding: 0;
```
add padding:0vh because there was too much spaces between btns in mobile

hrshkshri replied:
Maybe just remove this.

rahat2134 committed:
pass-lint

rahat2134 committed:
lint-checks

rahat2134 committed:
lint-checks

hrshkshri commented on core/templates/pages/about-page/about-page.component.css:
```diff
@@ -526,6 +537,15 @@
 }
+@media (max-width: 730px) {
+  .about-page .oppia-about-button {
+    padding: 3px;
```
Too small button ( as far as i can understand )

hrshkshri replied:
Can you attach a video of this change...

hrshkshri commented on core/templates/pages/about-page/about-page.component.css:
```diff
@@ -526,6 +537,15 @@
+    padding: 3px;
+  }
+  .oppia-about-page-section-one .oppia-about-title {
+    font-size: 23px;
```
`23px` is too small for heading... ?

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

nikitaevg reviewed (changes requested):
IMO this is too complicated for the task we're trying to solve - contain a button inside a div.

1. To solve this we use padding and font-size. Can't we use only one mechanism - font-size? It would be much simpler to understand.
2. Will your fix work correctly for different screen heights? I tested production, it doesn't work well when I shrink my screen height.

I see the following more clear approaches to fix this:
1. Use only font-size, it's simpler. But this doesn't fix the height issue.
2. Make the font size depend on the container size, like in [this](https://www.geeksforgeeks.org/font-scaling-based-on-width-of-container-using-css/) article.
3. Make the top block with "Get Started with Oppia" flexible, so that it changes its height if needed. This would require an approval from some designer or Sean.

I have little experience in html/css, there might be other approaches.

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 commented:
@nikitaevg I think width matters more than height, so we should focus on that first. I am working on adjusting the padding and font size because at certain widths, the padding for the oppia-about-title seems too large and looks odd. 

For eg : <img width="894" alt="Screenshot 2024-03-10 at 5 12 14 PM" src="https://github.com/oppia/oppia/assets/136263179/8b15aac4-e74a-4242-b032-c2d4dba77c85">

To make it look even, I adjusted the padding and changed the font size. We can't solely rely on padding to solve the issue, so changing the font size seemed necessary.

One possible solution for the height issue is to set the overflow property of the inner div to scroll once it exceeds a certain height.(May be a solution).

Thanks For your suggestion.

nikitaevg commented:
> I think width matters more than height

@rahat2134, it's not about what matters more, it's about a universal solution for any height. The numbers 960px, 868px and 698px work with your height, will they work for a different screen height?

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 commented:
> > I think width matters more than height
> 
> @rahat2134, it's not about what matters more, it's about a universal solution for any height. The numbers 960px, 868px and 698px work with your height, but will they work for a different screen height?


@nikitaevg , I have tried another solution for `overflow: scroll` as I mentioned earlier. Please take a look at the video -> (We just need to add overflow: scroll without changing any padding or anything).
[screen-capture (21).webm](https://github.com/oppia/oppia/assets/136263179/51851af1-c5a3-4670-9400-4d41eaf7bbb0)

nikitaevg commented:
> @nikitaevg , I have tried another solution for `overflow: scroll` as I mentioned earlier. Please take a look at the video -> (We just need to add overflow: scroll without changing any padding or anything). [screen-capture (21).webm](https://github.com/oppia/oppia/assets/136263179/51851af1-c5a3-4670-9400-4d41eaf7bbb0)

I don't think it's a good approach - it requires a user to scroll down to see a button, it's a _very_ counter intuitive thing to scroll.

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow

rahat2134 committed:
Merge branch 'develop' into About-page-overflow
